Test/Mocks/ClientMock.php

```diff
@@ -0,0 +1,259 @@
+<?php
```

Comment:
Could this class be cleaned up a little? I think if this is just a mock then it would only need to implement the methods in the ClientInterface and would not have to implement all the private methods. I think also it could be simplified by having the other resource methods (postResource, etc) just return the mock response how your getResource method works.

Also, would it be more accurate to name/describe this class as a stub instead of a mock? Since it is not observing the internal behavior and is just providing test data?